components/builder-datastore/hooks/install

```diff
@@ -0,0 +1,506 @@
+#!/bin/bash
+# shellcheck disable=SC1083
+# handlebars templating triggers SC1083
+
+exec 2>&1
+set -uo pipefail
+
+mkdir -p {{pkg.svc_path}} || exit 10
+cd {{pkg.svc_path}} || exit 11
+
+readonly OIFS=$IFS
+IFS='.'
+readonly PG_VERSION_FILE={{pkg.svc_data_path}}/PG_VERSION
+MOMENT="$(date '+%Y%m%d%H%M%S')"
+readonly MOMENT
+readonly PWFILE="{{pkg.svc_config_path}}/pwfile"
+readonly PACKAGED_DATASTORE_IDENT={{pkg.ident}}
+ACTIVE_IDENT=$PACKAGED_DATASTORE_IDENT
+readonly PG_TIMEOUT=30
+
+DEBUG() {
+  [[ $- == *x* ]]
+}
+
+set_installed_pg_ident_and_active_ident() {
+  if [[ -f $PG_VERSION_FILE ]]; then
+    local installed_pg_version
+    installed_pg_version=$(cat $PG_VERSION_FILE | sed 's/\.//')
+    INSTALLED_PG_IDENT="core/postgresql${installed_pg_version}"
+    sudo -E hab pkg install "$INSTALLED_PG_IDENT"
```

Comment:
we should probably only do this install if an upgrade is necessary. Otherwise it is possible we will end up with multiple versions of postgresql17 installed.

Reply:
I can add in some sort of guard in for but if a new version of postgres is promoted to on-prem-stable don't we want to start using that?  The part that is "version sensitive" is the postgres data files and those remain stable for each major release and the upgrade handles the case that we upgrade to a new major version.

Reply:
After discussing in a call will move the install command inside of the upgrade process to ensure that will only install and make available the old version of postgres we are upgrading from.

Reply:
Moved with a guard around it as discussed.